src/RecoObjects/Particle.cpp

```diff
@@ -141,6 +141,14 @@ Particle & Particle::operator =(const Particle &rightHandSide) {
 	return *this;
 }
 
+Particle & Particle::operator +=(const Particle &rightHandSide) {
+	fourvector = this->getFourVector() + rightHandSide.getFourVector();
+	// set mass to 0 since only dynamic mass makes sense from here on
```

Comment:
You can have mass in two ways:
- static (Particle::setMass(12344234))
- dynamic (four_vector.M()) aka from the 4-momentum